### PR TITLE
[#795] Projectify frontend part

### PR DESCRIPTION
**Pull request by gerhean**
Working on this.

Up to date with master branch as of 29/6/2021.

Things to do in another PR: 
- Add linter for pug
- Resolve eslint warnings
- Remove all `window.`

How to port code into new-frontend:
- Copy directly into `.vue` or same extension file of same name
  - `pug` file copy into \<template> portion
  - `js` file copy into \<script> portion
  - `css` file copy into \<style> portion 
- If there is `import aaa from bbb` at the top of the \<script> portion, search for `windows. utilsAaa` in the copied code and replace it with `aaa`
- For other exceptions may refer to the differences to the new-frontend file and old file

How to test:
1. First generate the desired report json using `gradlew run`
2.  (Working for windows, not tested on mac, not working on linux)
```
gradlew hotReloadFrontend
```

Build at `./reposense-report`:
```
gradlew run -Dargs="..."
```

Commit message:
```
Let's convert frontend to a standard Vue project.
```

### CODE REVIEW

**Review by fzdy1914 — CHANGES_REQUESTED** [review]
PLS split it into two PRs. One is to upgrade it to Vue3, another is projectify it.

You may do the first task directly in `frontend`. So, we can verify that it will not break anything.

**Comment by gerhean** [comment]
Is it not enough to just ensure that the new frontend is not broken? 

Vue3 completely breaks [vueify](https://github.com/vuejs/vueify) which the old frontend relies on. I'm not sure how to get around this.

One solution is to downgrade the new-frontend to Vue2, but feels redundant.

File: new-frontend/src/components/v-summary-charts.vue [hunk 1]
```diff
@@ -348,7 +348,7 @@ export default {
         zFromRamp: false,
         zFilterSearch: filterSearch,
       };
-      this.$emit('restoreZoomCommits', info);
+      this.$store.commit('updateTabZoomInfo', info);
```
Comment (fzdy1914) [inline new-frontend/src/components/v-summary-charts.vue:351]: This one should better be done in PR #1390. Do notice that, we are only changing related code with the topic of this PR. Pls only add necessary changes in this PR.

**Comment by gerhean** [comment]
Downgraded to Vue 2.6.12

**Comment by fzdy1914** [comment]
> Downgraded to Vue 2.6.12

So you prefer to use Vue 2.6 first and then upgrade to Vue 3? Any other difficulty you face in using Vue3 in the current project not considering the issue for vueify?

**Comment by gerhean** [comment]
> > Downgraded to Vue 2.6.12
> 
> So you prefer to use Vue 2.6 first and then upgrade to Vue 3? Any other difficulty you face in using Vue3 in the current project not considering the issue for vueify?

Some unexplainable errors which I find too hard to deal with.
image

**Comment by fzdy1914** [comment]
@gerhean Pls work on this PR sooner by my request. I want this PR to be merged by the end of this sem.

**Comment by gerhean** [comment]
I think the only thing left is to ask @dcshzj how to set up the integration tests.

**Comment by dcshzj** [comment]
You can modify build.gradle to point to the new frontend directory, assuming that it is a drop-in replacement. The report will then be generated using the new frontend and the cypress tests should be able to run automatically.

**Comment by fzdy1914** [comment]
Any more update? How is the deployment setup process going?

**Comment by gerhean** [comment]
@fzdy1914 do i replace the old deployment with the new one, or set up another one.

**Comment by gerhean** [comment]
Need #1527 to be reviewed and merged.

**Comment by damithc** [comment]
@gerhean this PR contains a lot of work. Any chance of wrapping it up?

**Comment by gerhean** [comment]
It is more or less done, I'll try to work on it tomorrow

**Comment by damithc** [comment]
> It is more or less done, I'll try to work on it tomorrow

Great. thanks @gerhean 
Ping @fzdy1914 when ready

**Comment by fzdy1914** [comment]
Done

**Comment by fzdy1914** [comment]
Can go ahead. @gerhean

**Comment by gerhean** [comment]
Should be mostly done, but not sure where to add documentation

**Comment by fzdy1914** [comment]
@gerhean Seems cannot find your new dashboard?

**Comment by gerhean** [comment]
Example of  new dashboard is at gerhean#2

image

Changes to github workflow files don't work until they are merged into master.

**Comment by fzdy1914** [comment]
@gerhean Yes, you are correct, forget this.

**Comment by fzdy1914** [comment]
@dcshzj Can you take a look at the gradlew/workflow part?

**Comment by fzdy1914** [comment]
The frontend part looks good to me.

**Comment by fzdy1914** [comment]
@dcshzj Re-Ping for review...

**Comment by dcshzj** [comment]
Apologies for missing this. With the updates to our gradle workflow, we will be adding additional tasks for building using the new frontend. However, my understanding is that this pull request will replace the old frontend entirely. Will this be the case, or will we be deprecating the old frontend only in the future?

**Comment by fzdy1914** [comment]
> Apologies for missing this. With the updates to our gradle workflow, we will be adding additional tasks for building using the new frontend. However, my understanding is that this pull request will replace the old frontend entirely. Will this be the case, or will we be deprecating the old frontend only in the future?

That's a very good question. In the beginning, I want the two frontend to be in the repo at the same time. So, the refactoring process can be easier and not breaking the master branch in case anything happens. Meanwhile, I originally plan to merge it at the half of prev semester, so we have another half of the semester to test it and finally remove the old one.

However, for now, actually, we can take the risk of potentially breaking the master as the new sem hasn't started yet.

@damithc Do you think we can replace the frontend directly and accept the risk(which is quite small)?

**Comment by damithc** [comment]
> However, for now, actually, we can take the risk of potentially breaking the master as the new sem hasn't started yet.
> 
> @damithc Do you think we can replace the frontend directly and accept the risk(which is quite small)?

Sure, go ahead @fzdy1914, if you are confident enough. It's fine as long as we fix any bugs before the semester starts.

**Comment by fzdy1914** [comment]
@gerhean Can you speed some time to refactor the code a bit to replace the frontend directly? Apologies for the inconvenience caused.

**Comment by gerhean** [comment]
I'll try when I'm free

**Comment by fzdy1914** [comment]
@gerhean Can you give a rough estimation? Can you finish it by 2 weeks?

**Comment by gerhean** [comment]
Surprisingly the checks all seem to pass and it works, but there's many warnings and outdated packages. Also the favicon seems to be missing, I'm not sure why it's not there, probably will fix soon.

**Comment by fzdy1914** [comment]
@gerhean This is a gentle reminder to continue your work. You got around 1 more weeks.

**Comment by gerhean** [comment]
@fzdy1914 I think its done, please help check. Also I'm not sure if it will be better to just add markbind to the package.json.

**Review by fzdy1914 — APPROVED** [review]
LGTM

**Pull request merged**